firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/SessionReportingCoordinator.java

```diff
@@ -44,6 +47,29 @@
   private static final int EVENT_THREAD_IMPORTANCE = 4;
   private static final int MAX_CHAINED_EXCEPTION_DEPTH = 8;
 
+  private static final int DEFAULT_MAX_EVENTS_TO_KEEP = 8;
+  private static final int DEFAULT_MAX_REPORTS_TO_KEEP = 4;
+
+  public static SessionReportingCoordinator create(
+      Context context,
+      IdManager idManager,
+      FileStore fileStore,
+      AppData appData,
+      LogFileManager logFileManager,
+      UserMetadata userMetadata,
+      StackTraceTrimmingStrategy stackTraceTrimmingStrategy) {
```

Comment:
How do we draw the line of which things get passed in vs made inside this create call? some things (e.g., this strategy) are created before the create call and injected in.)

Reply:
The things passed in this create call are all the underlying collaborators necessary to make the coordinator work in its default configuration (where 'configuration' is defined as the group of collaborators passed to the constructor). Most of these abstractions aren't perfect, so this create call is a bit messy. Would love to discuss this further if you have ideas about how to clean it up.

Reply:
Discussed this offline and decided this is fine for now.